packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -306,19 +302,86 @@ export class FeatureDevController {
                         },
                     ],
                 })
-            } else {
-                const errorMessage = createUserFacingErrorMessage(
-                    `${featureName} request failed: ${err.cause?.message ?? err.message}`
-                )
-                this.messenger.sendErrorMessage(
-                    errorMessage,
-                    message.tabID,
-                    this.retriesRemaining(session),
-                    session?.state.phase,
-                    session?.conversationIdUnsafe
-                )
+                break
+            default:
+                switch (session?.state.phase) {
+                    case DevPhase.APPROACH:
+                        if (isDenyListedError) {
+                            defaultMessage = `I'm sorry, I'm having trouble generating a plan. Please try again.`
+                        } else {
+                            defaultMessage = `I'm sorry, I ran into an issue while trying to approach the problem. Please try again.`
+                        }
+                        break
+                    case DevPhase.CODEGEN:
+                        if (isDenyListedError) {
+                            defaultMessage = `I'm sorry, I'm having trouble generating your code and can't continue at the moment. Please try again later, and share feedback to help me improve.`
+                            isSendErrorMessage = false
+                            this.messenger.sendAnswer({
```

Comment:
If we directly invoke `messenger.sendAnswer` here, we will not be able to show the sendFeedback button, as that logic exists inside `messenger.sendErrorMessage` [here](https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/amazonqFeatureDev/controllers/chat/messenger/messenger.ts#L68-L86)

Reply:
I've just added the sendFeedback for those scenarios.